### PR TITLE
Skip lib sources when rebuilding opened files

### PR DESCRIPTION
**Pull request by wclr**
Files that are opened in the editor automatically rebuilt (to show diagnostics without the need to update and save it), but it is not so actual for the library sources, esp considering that compiler [currently rebuilds if they are not changed ](https://github.com/purescript/purescript/issues/4066)(and rebuild is actually not needed to be rebuilt, that may unnecessarily trigger file watching tools). So this checks if the file path contains `.spago` or `bower_components` to skip it.

### CODE REVIEW

**Comment by nwolverson** [comment]
To be clear, this is just skipping triggering a rebuild on dependency files on open, it will still be triggered if one edits and saves the file? That seems like a good compromise, of course people may occasionally make quick edits to dependencies inside `.spago` for example and that should still show diagnostics.

File: src/LanguageServer/IdePurescript/Main.purs [hunk 1]
```diff
@@ -466,16 +472,19 @@ handleEvents config conn state documents logError = do
...
-      Just _ -> rebuildAndSendDiagnostics config conn state logError document
-      _ -> do
-        let uri = (un DocumentUri $ getUri document)
-        liftEffect $
-          Ref.modify_ (over ServerState (\st -> st
-          { buildQueue = Object.insert uri document (st.buildQueue)
-          })) state
+  onDidOpenDocument documents \{ document } ->
+    let uri = un DocumentUri $ getUri document
+    in case uri # isLibSourceFile of
```
Comment (nwolverson) [inline src/LanguageServer/IdePurescript/Main.purs:477]: how about making handler a `do` block and 
```
let uri = un DocumentUri $ getUri document
when (isLibSourceFile uri) $ launchAffLog do ...
```

Comment (wclr) [reply src/LanguageServer/IdePurescript/Main.purs:477]: Right, not used to use `when`, but it is absolutly appropriate here.

**Comment by wclr** [comment]
> To be clear, this is just skipping triggering a rebuild on dependency files on open, it will still be triggered if one edits and saves the file? That seems like a good compromise, of course people may occasionally make quick edits to dependencies inside `.spago` for example and that should still show diagnostics.

Absolutely, this branch only affects opened (activated) files, the next handler `onDidSaveDocument` responsible for pushing diagnostics when the file is changed and saved.

File: src/LanguageServer/IdePurescript/Main.purs (hunk 1 above)
Comment (wclr) [reply src/LanguageServer/IdePurescript/Main.purs]: Updated.

**Pull request merged**